client/src/app/components/client-gridview/client-gridview.component.html

```diff
@@ -4,25 +4,33 @@
       <!-- ID Column -->
       <ng-container matColumnDef="First Name">
         <th mat-header-cell *matHeaderCellDef mat-sort-header> First Name </th>
-        <td mat-cell *matCellDef="let row"> {{row.first_name}} </td>
+        <td mat-cell *matCellDef="let row" >
+          <span (keyup)="changeValue(row.id, 'first_name', $event)" (blur)="updateList(row.id, 'first_name', $event)" contenteditable="true">{{row.first_name}}</span>
+        </td>
       </ng-container>
   
       <!-- Progress Column -->
       <ng-container matColumnDef="Last Name">
         <th mat-header-cell *matHeaderCellDef mat-sort-header> Last Name </th>
-        <td mat-cell *matCellDef="let row"> {{row.last_name}} </td>
+        <td mat-cell *matCellDef="let row" >
+          <span (keyup)="changeValue(row.id, 'last_name', $event)" (blur)="updateList(row.id, 'last_name', $event)" contenteditable="true">{{row.last_name}}</span>
+        </td>
       </ng-container>
   
       <!-- Name Column -->
       <ng-container matColumnDef="Is Active">
         <th mat-header-cell *matHeaderCellDef mat-sort-header> Is Active </th>
-        <td mat-cell *matCellDef="let row"> {{row.is_active}} </td>
+        <td mat-cell *matCellDef="let row" >
+          <span (keyup)="changeValue(row.id, 'is_active', $event)" (blur)="updateList(row.id, 'is_active', $event)" contenteditable="true">{{row.is_active}}</span>
+        </td>
       </ng-container>
   
       <!-- Color Column -->
       <ng-container matColumnDef="Role">
         <th mat-header-cell *matHeaderCellDef mat-sort-header> Role </th>
-        <td mat-cell *matCellDef="let row" > {{row.role}} </td>
+        <td mat-cell *matCellDef="let row" >
+          <span (keyup)="changeValue(row.id, 'role', $event)" (blur)="updateList(row.id, 'role', $event)" contenteditable="true">{{row.role}}</span>
+        </td>
```

Comment:
Nice effort to make those fields editable, I can modify rows (with errors). In my opinion, this is not the best way to do it and I'll explain why. First, If I make errors while modifying those rows meaning that I add a value that is not allowed for example for a role i'm only allowed to be an SA, IM or SK, but you are allowing to modify any value there without throwing any error. Second, this is also happening for the true/flase Isactive column, it will change the value but this time if you refresh its throwing an error inside the console so it resets it. So my suggestion to make this thing better in terms of user experience, is that i would forget about that idea of modifying directly the table because this can lead to confusion and if as a developer you want to throw an error if fields are wrongly modified how would you do this in a clean way. So what I did in another project where I had to modify table rows that you could do is add a button on the right of each row called to update or you can simply add a click listener on the row, once clicked it will pop out a view card ( for example like the one I did for registration) that just holds the info of the row you clicked with the fields prepopulated and there you will be able to handle the errors more properly and for the fields that require specific options like the role or isactive you can add a dropdown menu for the list of choices.

Reply:
Look at the inconsistencies that i was able to create inside Is Active and Role:
![Screen Shot 2020-10-21 at 6 13 53 PM](https://user-images.githubusercontent.com/35696935/96792832-2f6da080-13c9-11eb-90d4-c4587805980a.png)